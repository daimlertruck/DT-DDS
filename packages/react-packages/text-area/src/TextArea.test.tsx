import { withProviders } from '@dt-ui/react-core';
import { act, render, screen, fireEvent } from '@testing-library/react';

import { TextArea } from '.';

describe('<TextArea /> component', () => {
  const ProvidedTextArea = withProviders(TextArea);

  it('renders input with an empty value', () => {
    const { getByRole } = render(
      <ProvidedTextArea label='My awesome text area' name='awesome-textarea' />
    );

    const textareInput = getByRole('textbox');

    expect(textareInput).toHaveValue('');
  });

  it('renders input with an value', () => {
    const label = 'My Input with prefilled value';
    const initialValue = 'Initial Value';

    const { container } = render(
      <ProvidedTextArea label={label} value={initialValue} />
    );

    const input = screen.getByLabelText(label);
    const value = screen.getByDisplayValue(initialValue);

    expect(value).toEqual(input);
    expect(container).toMatchSnapshot();
  });

  it('renders input with max length of 200 characters', () => {
    const { getByRole } = render(
      <ProvidedTextArea label='Awesome text area' maxLength={200} />
    );

    const textarea = getByRole('textbox');

    expect(textarea).toHaveProperty('maxLength', 200);
  });

  it('fills input correctly with new value on change Event', () => {
    const onChange = jest.fn();
    const { getByRole } = render(
      <ProvidedTextArea label='My awesome text area' onChange={onChange} />
    );

    const textareInput = getByRole('textbox');

    fireEvent.change(textareInput, { target: { value: 'New input value' } });

    expect(onChange).toHaveBeenCalled();
    expect(textareInput).toHaveValue('New input value');
  });

  it('should have active state on focus', () => {
    const { getByRole } = render(
      <ProvidedTextArea label='My awesome text area' />
    );

    const textarea = getByRole('textbox');

    act(() => {
      textarea.focus();
    });

    expect(textarea).toHaveStyle('outline: 1px solid #00677F');
    expect(textarea).toHaveStyle('border-width: 1px;');
    expect(textarea).toHaveStyle('border-style: solid');
    expect(textarea).toHaveStyle('border-color: transparent');
  });

  describe('onBlur event', () => {
    it("should add 'blur' with empty value", () => {
      const { getByRole } = render(
        <ProvidedTextArea label='My awesome text area' />
      );

      const textarea = getByRole('textbox');

      fireEvent.blur(textarea, { currentTarget: { value: '' } });

      expect(textarea).toHaveStyle('border: 1px solid #dedede');
    });

    it('should have active state', () => {
      const { getByRole, getByTestId } = render(
        <ProvidedTextArea label='Awesome text area' value='value' />
      );

      const textarea = getByRole('textbox');
      const label = getByTestId('label-field');

      fireEvent.blur(textarea, { currentTarget: { value: 'Some value' } });

      expect(label).toHaveStyle('font-size: 0.75rem');
      expect(label).toHaveStyle('transform: translateY(-45%)');
    });
  });
});
