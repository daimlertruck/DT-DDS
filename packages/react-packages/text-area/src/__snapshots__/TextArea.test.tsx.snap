// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> component should render default text area 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  font-family: Daimler CS;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: 0px;
  line-height: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #252525;
  cursor: inherit;
}

.emotion-2 {
  font-family: Daimler CS;
  font-size: 14px;
  font-weight: 400;
  letter-spacing: 0px;
  line-height: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #ffffff;
  min-height: 100px;
  width: 100%;
  outline: none;
  color: #252525;
  resize: none;
  padding: 12px;
  border: 0;
  margin-top: 8px;
  border-radius: 0px;
  border: 1px solid #c5c5c5;
}

.emotion-2::-webkit-input-placeholder {
  color: #3e3e3e;
}

.emotion-2::-moz-placeholder {
  color: #3e3e3e;
}

.emotion-2:-ms-input-placeholder {
  color: #3e3e3e;
}

.emotion-2::placeholder {
  color: #3e3e3e;
}

.emotion-2:focus-within,
.emotion-2:hover:not([disabled]) {
  border: 1px solid #00677f;
}

.emotion-2:-moz-read-only:not(:disabled) {
  border-color: #e6e6e6;
  color: #3e3e3e;
  background-color: #f2f2f2;
}

.emotion-2:read-only:not(:disabled) {
  border-color: #e6e6e6;
  color: #3e3e3e;
  background-color: #f2f2f2;
}

.emotion-2:-moz-read-only:not(:disabled):focus-within {
  border: 1px solid #00677f;
}

.emotion-2:-moz-read-only:not(:disabled):hover {
  border: 1px solid #00677f;
}

.emotion-2:read-only:not(:disabled):focus-within,
.emotion-2:read-only:not(:disabled):hover {
  border: 1px solid #00677f;
}

.emotion-2:disabled {
  cursor: not-allowed;
  color: #a9a9a9;
}

<div>
  <div
    class="emotion-0"
    style="position: relative; align-items: flex-start;"
  >
    <label
      class="emotion-1"
      data-testid="label-field"
      for="my-awesome-text-area-textarea"
      style="z-index: 1;"
    >
      My awesome text area
    </label>
    <textarea
      class="emotion-2 emotion-3"
      data-error="false"
      data-testid="my-awesome-text-area-textarea"
      id="my-awesome-text-area-textarea"
      name="awesome-textarea"
    >
      some value
    </textarea>
    <div
      class="emotion-0"
      style="flex-direction: row; justify-content: space-between; width: 100%; padding-inline: 12px; gap: 8px; margin-top: 4px;"
    />
  </div>
</div>
`;

exports[`<TextArea /> component should render text area with floating label 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  font-family: Daimler CS;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: 0px;
  line-height: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e3e3e;
  cursor: inherit;
  font-family: Daimler CS;
  font-size: 12px;
  font-weight: 400;
  letter-spacing: 0px;
  line-height: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  left: 12px;
  top: 16px;
  position: absolute;
  -webkit-transform: translateY(-45%);
  -moz-transform: translateY(-45%);
  -ms-transform: translateY(-45%);
  transform: translateY(-45%);
  -webkit-transition: -webkit-transform 150ms ease-out,font-size 150ms ease-out;
  transition: transform 150ms ease-out,font-size 150ms ease-out;
}

.emotion-2 {
  font-family: Daimler CS;
  font-size: 14px;
  font-weight: 400;
  letter-spacing: 0px;
  line-height: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #ffffff;
  min-height: 100px;
  width: 100%;
  outline: none;
  color: #252525;
  resize: none;
  padding: 24px 12px 12px 12px;
  border: 0;
  border-radius: 0px;
  border: 1px solid #c5c5c5;
}

.emotion-2::-webkit-input-placeholder {
  color: #3e3e3e;
}

.emotion-2::-moz-placeholder {
  color: #3e3e3e;
}

.emotion-2:-ms-input-placeholder {
  color: #3e3e3e;
}

.emotion-2::placeholder {
  color: #3e3e3e;
}

.emotion-2:focus-within,
.emotion-2:hover:not([disabled]) {
  border: 1px solid #00677f;
}

.emotion-2:-moz-read-only:not(:disabled) {
  border-color: #e6e6e6;
  color: #3e3e3e;
  background-color: #f2f2f2;
}

.emotion-2:read-only:not(:disabled) {
  border-color: #e6e6e6;
  color: #3e3e3e;
  background-color: #f2f2f2;
}

.emotion-2:-moz-read-only:not(:disabled):focus-within {
  border: 1px solid #00677f;
}

.emotion-2:-moz-read-only:not(:disabled):hover {
  border: 1px solid #00677f;
}

.emotion-2:read-only:not(:disabled):focus-within,
.emotion-2:read-only:not(:disabled):hover {
  border: 1px solid #00677f;
}

.emotion-2:disabled {
  cursor: not-allowed;
  color: #a9a9a9;
}

<div>
  <div
    class="emotion-0"
    style="position: relative; align-items: flex-start;"
  >
    <label
      class="emotion-1"
      data-testid="label-field"
      for="my-awesome-text-area-textarea"
      style="z-index: 1;"
    >
      My awesome text area
    </label>
    <textarea
      class="emotion-2 emotion-3"
      data-error="false"
      data-testid="my-awesome-text-area-textarea"
      id="my-awesome-text-area-textarea"
      name="awesome-textarea"
    >
      some value
    </textarea>
    <div
      class="emotion-0"
      style="flex-direction: row; justify-content: space-between; width: 100%; padding-inline: 12px; gap: 8px; margin-top: 4px;"
    />
  </div>
</div>
`;
